pkg/operator2/configsync.go

```diff
@@ -91,64 +91,189 @@ type sourceData struct {
 	mount  corev1.VolumeMount
 }
 
+// TODO: new source data could be generalized
+func newSourceDataIDPSecret(index int, secretName, idpType string) (string, sourceData) {
+	dest := getIDPName(index, secretName, idpType)
+	var ret sourceData
+
+	volume, mount, path := secretVolume(index, dest, idpType)
+	ret.src = secretName
+	ret.path = path
+	ret.volume = volume
+	ret.mount = mount
+
+	return dest, ret
+}
+
+func newSourceDataIDPConfigMap(index int, cmName, idpType string) (string, sourceData) {
+	dest := getIDPName(index, cmName, idpType)
+	var ret sourceData
+
+	volume, mount, path := secretVolume(index, dest, idpType)
+	ret.src = cmName
+	ret.path = path
+	ret.volume = volume
+	ret.mount = mount
+
+	return dest, ret
+}
+
 // TODO this should be combined with convertProviderConfigToOsinBytes as it would simplify how the data is shared
 func convertToData(idps []configv1.IdentityProvider) []idpSyncData {
 	out := make([]idpSyncData, 0, len(idps))
 	for i, idp := range idps {
 		pc := idp.ProviderConfig
 		switch pc.Type {
+		case configv1.IdentityProviderTypeBasicAuth:
+			p := pc.BasicAuth
+
+			ca := p.CA.Name
+			caDest, caData := newSourceDataIDPConfigMap(i, ca, corev1.ServiceAccountRootCAKey)
+
+			clientCert := p.TLSClientCert.Name
+			clientCertDest, clientCertData := newSourceDataIDPSecret(i, clientCert, corev1.TLSCertKey)
+
+			clientKey := p.TLSClientKey.Name
+			clientKeyDest, clienKeyData := newSourceDataIDPSecret(i, clientKey, corev1.TLSCertKey)
```

Comment:
Should be `corev1.TLSPrivateKeyKey`

Reply:
Good catch, I was afraid something might have gotten lost in those copy-pastes.